lib/ansible/module_utils/common/validation.py

```diff
@@ -0,0 +1,242 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.common.collections import is_iterable
+from ansible.module_utils.six import string_types
+
+
+def count_terms(terms, module_parameters):
+    """Count the number of occurrences of a key in a given dictionary
+
+    :arg terms: String or iterable of values to check
+    :arg module_parameters: Dictionary of module parameters
+
+    :returns: An integer that is the number of occurrences of the terms values
+        in the provided dictionary.
+    """
+
+    if not is_iterable(terms):
+        terms = [terms]
+
+    return len(set(terms).intersection(module_parameters))
+
+
+def check_mutually_exclusive(terms, module_parameters):
+    """Check mutually exclusive terms against argument parameters. Accepts
```

Comment:
I am split on my feelings about this function.  It doesn't work the same as it did as part of `AnsibleModule`.

Before, it would error on the first time that `count > 1`, whereas we aren't doing so now, we are collecting them all, but then not using them all.

Additionally, this is more like `list_mutually_exclusive`.  I think this function should raise some form of Exception, instead of just returning a list of the mutually exclusive args that were provided by the user.

Reply:
> Before, it would error on the first time that count > 1, whereas we aren't doing so now, 

I wanted the new check method to return complete results which may be useful to the caller. I will change how this output is presented in `fail_json()` so that we use all the results and give a better error message. That is a change in behavior, but I think it's for the better.

> I think this function should raise some form of Exception, instead of just returning a list of the mutually exclusive args that were provided by the user.

Agreed. I will create an `AnsibleModuleParameterError` exception that contains the results of the failed check.